pkg/middleware/recovery/recovery_test.go

```diff
@@ -0,0 +1,210 @@
+package recovery
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+	"google.golang.org/grpc/test/bufconn"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+
+	"github.com/openfga/openfga/pkg/logger"
+)
+
+func TestPanic(t *testing.T) {
+	logger := logger.MustNewLogger("text", "info", "unix")
+	handlerFunc := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		panic("Unexpected error!")
+	})
+
+	t.Run("With panic Recovery", func(t *testing.T) {
+		handler := Panic(handlerFunc, logger)
+
+		req, err := http.NewRequest(http.MethodGet, "/", nil)
+		if err != nil {
+			require.NoError(t, err)
+		}
+
+		resp := httptest.NewRecorder()
+		handler.ServeHTTP(resp, req)
```

Comment:
```suggestion
		require.NotPanics(t, func() {
            handler.ServeHTTP(resp, req)
        })
```

We want to make sure we don't panic, in addition to the response behavior.

Reply:
Done